pkgs/tools/audio/vosk-api/default.nix

```diff
@@ -0,0 +1,60 @@
+{ stdenv
+, fetchFromGitHub
+, fetchpatch
+, cmake
+, kaldi
+, openblas
+, python3
+, lib
+}:
+
+stdenv.mkDerivation rec {
+  name = "vosk-api";
+  version = "0.3.43";
+
+  src = fetchFromGitHub {
+    owner = "alphacep";
+    repo = "vosk-api";
+    rev = "v${version}";
+    sha256 = "SbCZx+G5q/W7CBVbmGZYf2ubhUJXtSTzaFAwQEJEt3Y=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+  ];
+
+  buildInputs = [
+    (kaldi.overrideAttrs (attrs: {
```

Comment:
Please move this to top-level (preferred) or in a let in

Reply:
I have yet to see applying patches in top-level. And do not see any benefit to extending the scope either.

Reply:
We also almost never apply patches in the inputs itself. Also the unpatched kaldi version could easily be used somewhere else in the file by accident.

Reply:
The only way to prevent that is to use shadowing. So I went there.

Reply:
Top-level overrides are now discouraged: https://github.com/NixOS/nixpkgs/issues/204303